src/Validation/ValidationURLProvider.php

```diff
@@ -0,0 +1,307 @@
+<?php
+/**
+ * Provides URLs to scan.
+ *
+ * @package AMP
+ * @since 2.1
+ */
+
+namespace AmpProject\AmpWP\Validation;
+
+use AMP_Theme_Support;
+use WP_Query;
+
+/**
+ * ValidationURLProvider class.
+ *
+ * @since 2.1
+ */
+final class ValidationURLProvider {
+
+	/**
+	 * Whether to include URLs that don't support AMP.
+	 *
+	 * @var bool
+	 */
+	public $include_unsupported;
+
+	/**
+	 * An allowlist of conditionals to use for querying URLs.
+	 *
+	 * Usually, this class will query all of the templates that don't have AMP disabled. This allows inclusion based on only these conditionals.
+	 *
+	 * @var array
+	 */
+	public $include_conditionals;
+
+	/**
+	 * The maximum number of URLs to provide for each content type.
+	 *
+	 * Templates are each a separate type, like those for is_category() and is_tag(), and each post type is a type.
+	 *
+	 * @var int
+	 */
+	public $limit_per_type;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param integer $limit_per_type The maximum number of URLs to validate for each type.
+	 * @param array   $include_conditionals An allowlist of conditionals to use for validation.
+	 * @param boolean $include_unsupported Whether to include URLs that don't support AMP.
+	 */
+	public function __construct(
+		$limit_per_type = 20,
+		$include_conditionals = [],
+		$include_unsupported = false
+	) {
+		$this->limit_per_type       = $limit_per_type;
+		$this->include_conditionals = $include_conditionals;
+		$this->include_unsupported  = $include_unsupported;
+	}
+
+	/**
+	 * Provides the array of URLs to check. Each URL is an array with two elements, with the URL at index 0 and the type at index 1.
```

Comment:
This long line can be split into short and long description.
```suggestion
	 * Provides the array of URLs to check.
	 *
	 * Each URL is an array with two elements, with the URL at index 0 and the type at index 1.
```